vscode-dotty/src/worksheet.ts

```diff
@@ -0,0 +1,346 @@
+'use strict'
+
+import * as vscode from 'vscode'
+import { LanguageClient } from 'vscode-languageclient'
+
+/** All decorations that have been added so far */
+let worksheetDecorationTypes: Map<vscode.TextDocument, vscode.TextEditorDecorationType[]> = new Map<vscode.TextDocument, vscode.TextEditorDecorationType[]>()
```

Comment:
I would make a class with fields instead of using global variables in this file.